File: src/VibeApp.API/Features/Vibes/Vibe.cs
using VibeApp.API.Shared.Common;

namespace VibeApp.API.Features.Vibes;

public class Vibe : BaseEntity
{
    public string Content { get; set; } = string.Empty;
    public Guid UserId { get; set; }
    public string? MediaUrl { get; set; }
    public VibeType Type { get; set; }
    public int LikesCount { get; set; }
    public int CommentsCount { get; set; }
    public int SharesCount { get; set; }
    public bool IsPublic { get; set; } = true;
}

public enum VibeType
{
    Text,
    Image,
    Video,
    Audio
}

File: frontend/src/types/vibe.ts
export enum VibeType {
  TEXT = 'text',
  IMAGE = 'image',
  VIDEO = 'video',
  AUDIO = 'audio',
}

export interface Vibe {
  id: string;
  content: string;
  user_id: string;
  media_url?: string;
  type: VibeType;
  likes_count: number;
  comments_count: number;
  shares_count: number;
  is_public: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateVibeRequest {
  content: string;
  user_id: string;
  media_url?: string;
  type: VibeType;
  is_public: boolean;
}
